sql/parser/eval.go

```diff
@@ -136,13 +139,13 @@ var binOps = map[binArgs]binOp{
 	binArgs{Plus, dateType, intervalType}: {
 		returnType: DummyTimestamp,
 		fn: func(left Datum, right Datum) (Datum, error) {
-			return DTimestamp{Time: left.(DDate).Add(right.(DInterval).Duration)}, nil
+			return DTimestamp{Time: left.(DDate).toTime().Add(right.(DInterval).Duration)}, nil
```

Comment:
I've taken out a few binary operations like timestamp +- date. Ready for another viewing. Thanks